### PR TITLE
TianhaoHU, Hw8

### PR DESCRIPTION
**Pull request by Whuloser**

### CODE REVIEW

File: courseworks/week8/TianhaoHU/hw8.ipynb [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    "            for j in (i+1):vertex_count\n",
+    "                xi, xj = x[2*i-1:2*i], x[2*j-1:2*j]\n",
+    "                dist = sqrt(sum((xi - xj).^2))\n",
+    "                connected = any([(i, j) == edge || (j, i) == edge for edge in edges])\n",
+    "\n",
+    "                if connected && dist > 1.0\n",
+    "                    total_loss += (dist - 1.0)^2\n",
+    "                elseif !connected && dist < 1.0\n",
+    "                    total_loss += (1.0 - dist)^2\n",
+    "                end\n",
```
Comment (ArrogantGao) [inline courseworks/week8/TianhaoHU/hw8.ipynb:41]: here you directly defined $ r = 1 $ as the condition, however for the unitdisk embeding I will recommand you that you may choos 0.95 as requirement for 'connected' and verse vice.

File: courseworks/week8/TianhaoHU/hw8.ipynb [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    "    res = optimize(loss_function, vec(locations), LBFGS())\n",
+    "    min_loss = Optim.minimum(res)\n",
+    "\n",
+    "    if min_loss < 1e-6\n",
+    "        locations_opt = reshape(Optim.minimizer(res), 2, vertex_count)\n",
+    "        return true, locations_opt\n",
+    "    else\n",
+    "        return false, nothing\n",
+    "    end\n",
+    "end"
```
Comment (ArrogantGao) [inline courseworks/week8/TianhaoHU/hw8.ipynb:77]: Here you only checked if the loss function has beed optimized, but you also need to check by the defination of the unitdisk embeding, becase they are not equivalent.

**Review by ArrogantGao — CHANGES_REQUESTED** [review]
